core/include/scipp/core/variable.h

```diff
@@ -755,10 +777,21 @@ class SCIPP_CORE_EXPORT VariableProxy : public VariableConstProxy {
     m_view = slice.data().makeView(dim, begin, end);
   }
 
+  template <class Var>
+  static VariableProxy makeTransposed(Var &var,
+                                      const std::vector<Dim> &dimOrder) {
+    auto res = VariableProxy(var);
+    auto tmpConstProxy = VariableConstProxy(var);
+    res.m_view = tmpConstProxy.data().transpose(dimOrder);
```

Comment:
Isn't the view you are creating here const, i.e., attempts to modify an element would fail?